<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>966</width>
    <height>578</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Flock</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="s_mainWindowGridLayout">
    <item row="0" column="0">
     <spacer name="horizontalSpacer">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>40</width>
        <height>20</height>
       </size>
      </property>
     </spacer>
    </item>
    <item row="0" column="2">
     <widget class="QTabWidget" name="m_optionsTabWidget">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <property name="currentIndex">
       <number>0</number>
      </property>
      <property name="usesScrollButtons">
       <bool>true</bool>
      </property>
      <property name="documentMode">
       <bool>false</bool>
      </property>
      <property name="tabsClosable">
       <bool>false</bool>
      </property>
      <property name="movable">
       <bool>false</bool>
      </property>
      <property name="tabBarAutoHide">
       <bool>false</bool>
      </property>
      <widget class="QWidget" name="m_flockTab">
       <property name="toolTip">
        <string/>
       </property>
       <attribute name="title">
        <string>Flock</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_4">
        <item row="1" column="0">
         <widget class="QGroupBox" name="m_BehaviourOptions">
          <property name="title">
           <string>Boid Behaviours</string>
          </property>
          <layout class="QGridLayout" name="gridLayout">
           <item row="5" column="0">
            <widget class="QCheckBox" name="m_fov">
             <property name="text">
              <string>Field of View</string>
             </property>
            </widget>
           </item>
           <item row="1" column="2">
            <widget class="QSlider" name="m_separation">
             <property name="maximum">
              <number>500</number>
             </property>
             <property name="value">
              <number>100</number>
             </property>
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
            </widget>
           </item>
           <item row="2" column="2">
            <widget class="QSlider" name="m_alignment">
             <property name="maximum">
              <number>500</number>
             </property>
             <property name="value">
              <number>50</number>
             </property>
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
            </widget>
           </item>
           <item row="0" column="0">
            <widget class="QLabel" name="m_cohesionLabel">
             <property name="text">
              <string>Cohesion</string>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="m_spearatioLabel">
             <property name="text">
              <string>Separation</string>
             </property>
            </widget>
           </item>
           <item row="2" column="0">
            <widget class="QLabel" name="m_alignLabel">
             <property name="text">
              <string>Alignment</string>
             </property>
            </widget>
           </item>
           <item row="3" column="0">
            <widget class="QLabel" name="m_speedLabel">
             <property name="text">
              <string>Speed</string>
             </property>
            </widget>
           </item>
           <item row="3" column="2">
            <widget class="QSlider" name="m_speed">
             <property name="maximum">
              <number>50</number>
             </property>
             <property name="singleStep">
              <number>1</number>
             </property>
             <property name="value">
              <number>5</number>
             </property>
             <property name="sliderPosition">
              <number>5</number>
             </property>
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
            </widget>
           </item>
           <item row="0" column="2">
            <widget class="QSlider" name="m_cohesion">
             <property name="maximum">
              <number>500</number>
             </property>
             <property name="value">
              <number>200</number>
             </property>
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
            </widget>
           </item>
           <item row="6" column="2">
            <widget class="QSpinBox" name="m_fovAngle">
             <property name="maximum">
              <number>180</number>
             </property>
             <property name="value">
              <number>100</number>
             </property>
            </widget>
           </item>
           <item row="6" column="0">
            <widget class="QLabel" name="m_fovAngleLabel">
             <property name="text">
              <string>Field of View Angle</string>
             </property>
            </widget>
           </item>
           <item row="4" column="0">
            <widget class="QLabel" name="m_massLabel">
             <property name="text">
              <string>Mass</string>
             </property>
            </widget>
           </item>
           <item row="4" column="2">
            <widget class="QSlider" name="m_mass">
             <property name="minimum">
              <number>1</number>
             </property>
             <property name="maximum">
              <number>30</number>
             </property>
             <property name="value">
              <number>15</number>
             </property>
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
            </widget>
           </item>
          </layout>
          <zorder>m_cohesion</zorder>
          <zorder>m_separation</zorder>
          <zorder>m_alignment</zorder>
          <zorder>m_cohesionLabel</zorder>
          <zorder>m_spearatioLabel</zorder>
          <zorder>m_alignLabel</zorder>
          <zorder>m_speedLabel</zorder>
          <zorder>m_speed</zorder>
          <zorder>m_fov</zorder>
          <zorder>m_fovAngleLabel</zorder>
          <zorder>m_fovAngle</zorder>
          <zorder>m_massLabel</zorder>
          <zorder>m_mass</zorder>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QGroupBox" name="m_flockOptions">
          <property name="title">
           <string>Flock</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_3">
           <item row="0" column="0">
            <widget class="QPushButton" name="m_addBoid">
             <property name="text">
              <string>Add Boid</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QCheckBox" name="m_leader">
             <property name="text">
              <string>Leader</string>
             </property>
            </widget>
           </item>
           <item row="3" column="1">
            <widget class="QPushButton" name="m_newFlock">
             <property name="text">
              <string>Create New Flock</string>
             </property>
            </widget>
           </item>
           <item row="3" column="0">
            <widget class="QSpinBox" name="m_flockCount">
             <property name="minimum">
              <number>1</number>
             </property>
             <property name="maximum">
              <number>500</number>
             </property>
             <property name="value">
              <number>1</number>
             </property>
            </widget>
           </item>
           <item row="2" column="1">
            <widget class="QCheckBox" name="m_predator">
             <property name="text">
              <string>Predator</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QCheckBox" name="m_steer">
             <property name="text">
              <string>Steerable Leader</string>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QPushButton" name="m_removeBoid">
             <property name="text">
              <string>Remove Boid</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="m_obstacleTab">
       <attribute name="title">
        <string>Obstacle</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_5">
        <item row="1" column="0">
         <widget class="QGroupBox" name="m_presetObstacleOptions">
          <property name="title">
           <string>Preset Courses</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_7">
           <item row="0" column="0">
            <widget class="QComboBox" name="m_presetObstacles">
             <property name="currentText">
              <string>None</string>
             </property>
             <item>
              <property name="text">
               <string>None</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Crecents</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Horn</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Boromean Rings</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Wavey Tunnel</string>
              </property>
             </item>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QGroupBox" name="m_customObstacleOptions">
          <property name="title">
           <string>Custom Obstacles</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_2">
           <item row="6" column="0">
            <widget class="QLabel" name="obstacleradius">
             <property name="text">
              <string>Obstacle Radius</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QComboBox" name="m_customObstacle">
             <property name="maxCount">
              <number>10</number>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="label">
             <property name="text">
              <string>Custom Obstacle number</string>
             </property>
            </widget>
           </item>
           <item row="3" column="0" colspan="2">
            <layout class="QHBoxLayout" name="horizontalLayout_2">
             <property name="topMargin">
              <number>0</number>
             </property>
             <item>
              <widget class="QLabel" name="label_3">
               <property name="text">
                <string>End point (x,y,z</string>
               </property>
              </widget>
             </item>
             <item>
              <spacer name="horizontalSpacer_4">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>20</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
             <item>
              <widget class="QDoubleSpinBox" name="m_customEndX">
               <property name="minimum">
                <double>-1000.000000000000000</double>
               </property>
               <property name="maximum">
                <double>1000.000000000000000</double>
               </property>
               <property name="singleStep">
                <double>2.000000000000000</double>
               </property>
               <property name="value">
                <double>0.000000000000000</double>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QDoubleSpinBox" name="m_customEndY">
               <property name="minimum">
                <double>-1000.000000000000000</double>
               </property>
               <property name="maximum">
                <double>1000.000000000000000</double>
               </property>
               <property name="singleStep">
                <double>2.000000000000000</double>
               </property>
               <property name="value">
                <double>-150.000000000000000</double>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QDoubleSpinBox" name="m_customEndZ">
               <property name="minimum">
                <double>-1000.000000000000000</double>
               </property>
               <property name="maximum">
                <double>1000.000000000000000</double>
               </property>
               <property name="singleStep">
                <double>2.000000000000000</double>
               </property>
               <property name="value">
                <double>-100.000000000000000</double>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item row="2" column="0" colspan="2">
            <layout class="QHBoxLayout" name="horizontalLayout">
             <property name="topMargin">
              <number>0</number>
             </property>
             <item>
              <widget class="QLabel" name="label_2">
               <property name="text">
                <string>Start point (x,y,z)</string>
               </property>
              </widget>
             </item>
             <item>
              <spacer name="horizontalSpacer_5">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>10</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
             <item>
              <widget class="QDoubleSpinBox" name="m_customStartX">
               <property name="minimum">
                <double>-1000.000000000000000</double>
               </property>
               <property name="maximum">
                <double>1000.000000000000000</double>
               </property>
               <property name="singleStep">
                <double>2.000000000000000</double>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QDoubleSpinBox" name="m_customStartY">
               <property name="minimum">
                <double>-1000.000000000000000</double>
               </property>
               <property name="maximum">
                <double>1000.000000000000000</double>
               </property>
               <property name="singleStep">
                <double>2.000000000000000</double>
               </property>
               <property name="value">
                <double>-150.000000000000000</double>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QDoubleSpinBox" name="m_customStartZ">
               <property name="minimum">
                <double>-1000.000000000000000</double>
               </property>
               <property name="maximum">
                <double>1000.000000000000000</double>
               </property>
               <property name="singleStep">
                <double>2.000000000000000</double>
               </property>
               <property name="value">
                <double>100.000000000000000</double>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item row="4" column="0" colspan="2">
            <layout class="QHBoxLayout" name="horizontalLayout_3">
             <property name="topMargin">
              <number>0</number>
             </property>
             <item>
              <widget class="QLabel" name="label_4">
               <property name="text">
                <string>Control Point (x,y,z)</string>
               </property>
              </widget>
             </item>
             <item>
              <spacer name="horizontalSpacer_6">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>10</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
             <item>
              <widget class="QDoubleSpinBox" name="m_customCtrlX">
               <property name="minimum">
                <double>-1000.000000000000000</double>
               </property>
               <property name="maximum">
                <double>1000.000000000000000</double>
               </property>
               <property name="singleStep">
                <double>2.000000000000000</double>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QDoubleSpinBox" name="m_customCtrlY">
               <property name="minimum">
                <double>-1000.000000000000000</double>
               </property>
               <property name="maximum">
                <double>1000.000000000000000</double>
               </property>
               <property name="singleStep">
                <double>2.000000000000000</double>
               </property>
               <property name="value">
                <double>100.000000000000000</double>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QDoubleSpinBox" name="m_customCtrlZ">
               <property name="minimum">
                <double>-1000.000000000000000</double>
               </property>
               <property name="maximum">
                <double>1000.000000000000000</double>
               </property>
               <property name="singleStep">
                <double>2.000000000000000</double>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item row="0" column="1">
            <widget class="QPushButton" name="m_removeCustomObstacle">
             <property name="text">
              <string>Remove Custom Obstacle</string>
             </property>
            </widget>
           </item>
           <item row="0" column="0">
            <widget class="QPushButton" name="m_addCustomObstacle">
             <property name="text">
              <string>Add Custom Obstacle</string>
             </property>
            </widget>
           </item>
           <item row="8" column="0">
            <widget class="QCheckBox" name="m_showCtrlPoints">
             <property name="text">
              <string>Show Control Point</string>
             </property>
            </widget>
           </item>
           <item row="6" column="1">
            <widget class="QSpinBox" name="m_customObsRad">
             <property name="value">
              <number>10</number>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QGroupBox" name="m_obstacleOptions">
          <property name="title">
           <string>Obstacles</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_6">
           <item row="1" column="0" colspan="2">
            <widget class="QPushButton" name="m_clearObstacle">
             <property name="text">
              <string>Clear Obstacles</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QPushButton" name="m_removeObstacle">
             <property name="text">
              <string>Remove Obstacle</string>
             </property>
            </widget>
           </item>
           <item row="0" column="0">
            <widget class="QPushButton" name="m_addObstacle">
             <property name="text">
              <string>Add Obstacle</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item row="3" column="0">
         <spacer name="verticalSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="m_visTab">
       <attribute name="title">
        <string>Visualisation</string>
       </attribute>
       <layout class="QFormLayout" name="formLayout">
        <item row="0" column="1">
         <widget class="QCheckBox" name="m_visOctree">
          <property name="text">
           <string>Visualise Octree</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QCheckBox" name="m_tails">
          <property name="text">
           <string>Trails</string>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QCheckBox" name="m_pov">
          <property name="text">
           <string>Point of View</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
    <item row="0" column="1">
     <spacer name="horizontalSpacer_2">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>40</width>
        <height>20</height>
       </size>
      </property>
     </spacer>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>966</width>
     <height>20</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
